wx/lib/agw/scrolledthumbnail.py

```diff
@@ -0,0 +1,2121 @@
+        sel = self.GetItemIndex(x, y)
+
+        if sel == self._pointed:
+            if self._enabletooltip and sel >= 0:
+                if not hasattr(self, "_tipwindow"):
+                    self._tipwindow = wx.ToolTip(self.GetThumbInfo(sel))
+                    self._tipwindow.SetDelay(1000)
+                    self.SetToolTip(self._tipwindow)
+                else:
+                    self._tipwindow.SetDelay(1000)
+                    self._tipwindow.SetTip(self.GetThumbInfo(sel))
+
+            event.Skip()
+            return
+
+        if self._enabletooltip:
+            if hasattr(self, "_tipwindow"):
+                self._tipwindow.Enable(False)
+
+        # update thumbnail
+        self._pointed = sel
+
+        if self._enabletooltip and sel >= 0:
+            if not hasattr(self, "_tipwindow"):
+                self._tipwindow = wx.ToolTip(self.GetThumbInfo(sel))
+                self._tipwindow.SetDelay(1000)
+                self._tipwindow.Enable(True)
+                self.SetToolTip(self._tipwindow)
+            else:
+                self._tipwindow.SetDelay(1000)
+                self._tipwindow.Enable(True)
+                self._tipwindow.SetTip(self.GetThumbInfo(sel))
+
+        self.Refresh()
+        eventOut = ThumbnailEvent(wxEVT_THUMBNAILS_POINTED, self.GetId())
+        self.GetEventHandler().ProcessEvent(eventOut)
+        event.Skip()
+
+
+    def OnMouseLeave(self, event):
+        """
+        Handles the ``wx.EVT_LEAVE_WINDOW`` event for :class:`ThumbnailCtrl`.
+
+        :param `event`: a :class:`MouseEvent` event to be processed.
+        """
+
+        if self._pointed != -1:
+
+            self._pointed = -1
+            self.Refresh()
+            eventOut = ThumbnailEvent(wxEVT_THUMBNAILS_POINTED, self.GetId())
+            self.GetEventHandler().ProcessEvent(eventOut)
+
+
+    def OnThumbChanged(self, event):
+        """
+        Handles the ``EVT_THUMBNAILS_THUMB_CHANGED`` event for :class:`ThumbnailCtrl`.
+
+        :param `event`: a :class:`ThumbnailEvent` event to be processed.
+        """
+
+        for ii in range(len(self._items)):
+            if self._items[ii].GetFileName() == event.GetString():
+
+                self._items[ii].SetFilename(self._items[ii].GetFileName())
+                if event.GetClientData():
+
+                    img = wx.Image(event.GetClientData())
+                    self._items[ii].SetImage(img)
+
+        self.Refresh()
+
+
+    def OnChar(self, event):
+        """
+        Handles the ``wx.EVT_CHAR`` event for :class:`ThumbnailCtrl`.
+
+        :param `event`: a :class:`KeyEvent` event to be processed.
+
+        :note: You have these choices:
+
+         (1) ``d`` key rotates 90 degrees clockwise the selected thumbnails;
+         (2) ``s`` key rotates 90 degrees counter-clockwise the selected thumbnails;
+         (3) ``a`` key rotates 180 degrees the selected thumbnails;
+         (4) ``Del`` key deletes the selected thumbnails;
```

Comment:
My eyes don't see this in the evt handler... What is this supposed to do...?
Delete the thumb from the widget, Delete the file associated with the thumb(BAD in python this is unrecoverable) or send associated file to the RecycleBin(possible better choice, recoverable)...?
Better explanation needed...